// ngx-bad-phone-spinner/esm5/index.js
import * as tslib_1 from "tslib";
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgxBadPhoneSpinnerComponent } from './ngx-bad-phone-spinner/ngx-bad-phone-spinner.component';
import { RangePipe } from './pipes/range.pipe';
export * from './models/ngx-bad-phone-spinner.model';
export * from './models/lock-options.model';
export * from './ngx-bad-phone-spinner/ngx-bad-phone-spinner.component';
var NgxBadPhoneSpinnerModule = /** @class */ (function () {
    function NgxBadPhoneSpinnerModule() {
    }
    NgxBadPhoneSpinnerModule_1 = NgxBadPhoneSpinnerModule;
    NgxBadPhoneSpinnerModule.forRoot = function () {
        return {
            ngModule: NgxBadPhoneSpinnerModule_1,
            providers: []
        };
    };
    var NgxBadPhoneSpinnerModule_1;
    NgxBadPhoneSpinnerModule = NgxBadPhoneSpinnerModule_1 = tslib_1.__decorate([
        NgModule({
            declarations: [
                NgxBadPhoneSpinnerComponent,
                RangePipe
            ],
            exports: [
                NgxBadPhoneSpinnerComponent,
                RangePipe
            ],
            imports: [
                CommonModule,
                FormsModule
            ],
            providers: []
        })
    ], NgxBadPhoneSpinnerModule);
    return NgxBadPhoneSpinnerModule;
}());
export { NgxBadPhoneSpinnerModule };
