// ngx-bad-phone-spinner/bundles/ngx-bad-phone-spinner.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common'), require('@angular/forms')) :
    typeof define === 'function' && define.amd ? define('ngx-bad-phone-spinner', ['exports', '@angular/core', '@angular/common', '@angular/forms'], factory) :
    (global = global || self, factory(global['ngx-bad-phone-spinner'] = {}, global.ng.core, global.ng.common, global.ng.forms));
}(this, (function (exports, core, common, forms) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    /* global Reflect, Promise */

    var extendStatics = function(d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };

    function __extends(d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    }

    var __assign = function() {
        __assign = Object.assign || function __assign(t) {
            for (var s, i = 1, n = arguments.length; i < n; i++) {
                s = arguments[i];
                for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p)) t[p] = s[p];
            }
            return t;
        };
        return __assign.apply(this, arguments);
    };

    function __rest(s, e) {
        var t = {};
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
            t[p] = s[p];
        if (s != null && typeof Object.getOwnPropertySymbols === "function")
            for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
                if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                    t[p[i]] = s[p[i]];
            }
        return t;
    }

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
            function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }

    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f) throw new TypeError("Generator is already executing.");
            while (_) try {
                if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
                if (y = 0, t) op = [op[0] & 2, t.value];
                switch (op[0]) {
                    case 0: case 1: t = op; break;
                    case 4: _.label++; return { value: op[1], done: false };
                    case 5: _.label++; y = op[1]; op = [0]; continue;
                    case 7: op = _.ops.pop(); _.trys.pop(); continue;
                    default:
                        if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                        if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                        if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                        if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                        if (t[2]) _.ops.pop();
                        _.trys.pop(); continue;
                }
                op = body.call(thisArg, _);
            } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
            if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    function __exportStar(m, exports) {
        for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    function __read(o, n) {
        var m = typeof Symbol === "function" && o[Symbol.iterator];
        if (!m) return o;
        var i = m.call(o), r, ar = [], e;
        try {
            while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
        }
        catch (error) { e = { error: error }; }
        finally {
            try {
                if (r && !r.done && (m = i["return"])) m.call(i);
            }
            finally { if (e) throw e.error; }
        }
        return ar;
    }

    function __spread() {
        for (var ar = [], i = 0; i < arguments.length; i++)
            ar = ar.concat(__read(arguments[i]));
        return ar;
    }

    function __spreadArrays() {
        for (var s = 0, i = 0, il = arguments.length; i < il; i++) s += arguments[i].length;
        for (var r = Array(s), k = 0, i = 0; i < il; i++)
            for (var a = arguments[i], j = 0, jl = a.length; j < jl; j++, k++)
                r[k] = a[j];
        return r;
    };

    function __await(v) {
        return this instanceof __await ? (this.v = v, this) : new __await(v);
    }

    function __asyncGenerator(thisArg, _arguments, generator) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var g = generator.apply(thisArg, _arguments || []), i, q = [];
        return i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i;
        function verb(n) { if (g[n]) i[n] = function (v) { return new Promise(function (a, b) { q.push([n, v, a, b]) > 1 || resume(n, v); }); }; }
        function resume(n, v) { try { step(g[n](v)); } catch (e) { settle(q[0][3], e); } }
        function step(r) { r.value instanceof __await ? Promise.resolve(r.value.v).then(fulfill, reject) : settle(q[0][2], r); }
        function fulfill(value) { resume("next", value); }
        function reject(value) { resume("throw", value); }
        function settle(f, v) { if (f(v), q.shift(), q.length) resume(q[0][0], q[0][1]); }
    }

    function __asyncDelegator(o) {
        var i, p;
        return i = {}, verb("next"), verb("throw", function (e) { throw e; }), verb("return"), i[Symbol.iterator] = function () { return this; }, i;
        function verb(n, f) { i[n] = o[n] ? function (v) { return (p = !p) ? { value: __await(o[n](v)), done: n === "return" } : f ? f(v) : v; } : f; }
    }

    function __asyncValues(o) {
        if (!Symbol.asyncIterator) throw new TypeError("Symbol.asyncIterator is not defined.");
        var m = o[Symbol.asyncIterator], i;
        return m ? m.call(o) : (o = typeof __values === "function" ? __values(o) : o[Symbol.iterator](), i = {}, verb("next"), verb("throw"), verb("return"), i[Symbol.asyncIterator] = function () { return this; }, i);
        function verb(n) { i[n] = o[n] && function (v) { return new Promise(function (resolve, reject) { v = o[n](v), settle(resolve, reject, v.done, v.value); }); }; }
        function settle(resolve, reject, d, v) { Promise.resolve(v).then(function(v) { resolve({ value: v, done: d }); }, reject); }
    }

    function __makeTemplateObject(cooked, raw) {
        if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
        return cooked;
    };

    function __importStar(mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
        result.default = mod;
        return result;
    }

    function __importDefault(mod) {
        return (mod && mod.__esModule) ? mod : { default: mod };
    }

    (function (LockOptions) {
        LockOptions["IGNORE"] = "ignore";
        LockOptions["LIAR"] = "liar";
        LockOptions["MANUAL"] = "manual";
        LockOptions["ORDER"] = "order";
        LockOptions["RANDOM"] = "random";
        LockOptions["REVERSE"] = "reverse";
        LockOptions["SHUFFLE"] = "shuffle";
    })(exports.LockOptions || (exports.LockOptions = {}));

    var NgxBadPhoneSpinnerOptions = /** @class */ (function () {
        function NgxBadPhoneSpinnerOptions() {
            this.locks = [
                exports.LockOptions.MANUAL
            ];
            this.unlocks = [
                exports.LockOptions.MANUAL
            ];
            this.shuffleMilliseconds = 200;
            this.shufflesPerClick = 5;
        }
        return NgxBadPhoneSpinnerOptions;
    }());

    var NgxBadPhoneSpinnerComponent = /** @class */ (function () {
        function NgxBadPhoneSpinnerComponent() {
            this._propagateChange = function () { };
            this.fullNumber = '0000000000';
            this.digits = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
            this.locks = [false, false, false, false, false, false, false, false, false, false];
            this.lockIcons = [false, false, false, false, false, false, false, false, false, false];
            this.lockColors = [false, false, false, false, false, false, false, false, false, false];
            this.disabled = false;
            this.isRandomizing = false;
            this.options = new NgxBadPhoneSpinnerOptions();
            this.change = new core.EventEmitter();
        }
        NgxBadPhoneSpinnerComponent_1 = NgxBadPhoneSpinnerComponent;
        Object.defineProperty(NgxBadPhoneSpinnerComponent.prototype, "number", {
            set: function (digits) {
                var candidates = digits.split('');
                for (var i = 0; i < 10; i++) {
                    this.digits[i] = typeof candidates[i] !== 'undefined' ? parseInt(candidates[i], 10) : 0;
                }
            },
            enumerable: true,
            configurable: true
        });
        NgxBadPhoneSpinnerComponent.prototype.writeValue = function (value) {
            this.fullNumber = value || '0000000000';
        };
        NgxBadPhoneSpinnerComponent.prototype.registerOnChange = function (fn) {
            this._propagateChange = fn;
        };
        NgxBadPhoneSpinnerComponent.prototype.registerOnTouched = function (fn) {
            this._onTouched = fn;
        };
        NgxBadPhoneSpinnerComponent.prototype.setDisabledState = function (isDisabled) {
            this.disabled = isDisabled;
        };
        NgxBadPhoneSpinnerComponent.prototype.toggleLock = function (i) {
            var _this = this;
            var canChange = true;
            if (i !== 0) {
                if (this.options.locks.indexOf(exports.LockOptions.ORDER) !== -1) {
                    if (i !== 0) {
                        if (this.locks[i - 1] === false) {
                            canChange = false;
                        }
                    }
                }
                if (this.options.unlocks.indexOf(exports.LockOptions.ORDER) !== -1) {
                    if (i !== 0) {
                        if (this.locks[i - 1] === true) {
                            canChange = false;
                        }
                    }
                }
                if (this.options.locks.indexOf(exports.LockOptions.REVERSE) !== -1) {
                    if (i + 1 < this.locks.length) {
                        if (this.locks[i + 1] === false) {
                            canChange = false;
                        }
                    }
                }
                if (this.options.unlocks.indexOf(exports.LockOptions.REVERSE) !== -1) {
                    if (i + 1 < this.locks.length) {
                        if (this.locks[i + 1] === true) {
                            canChange = false;
                        }
                    }
                }
            }
            if (this.options.unlocks.indexOf(exports.LockOptions.IGNORE) !== -1 && Math.random() < 0.5) {
                canChange = false;
            }
            if (this.options.locks.indexOf(exports.LockOptions.IGNORE) !== -1 && Math.random() < 0.5) {
                canChange = false;
            }
            if (canChange) {
                this.locks[i] = !this.locks[i];
                if (this.locks[i]) {
                    if (this.options.unlocks.indexOf(exports.LockOptions.RANDOM) !== -1) {
                        setTimeout(function () {
                            _this.locks[i] = false;
                        }, Math.random() * 1000 * 60);
                    }
                }
                else {
                    if (this.options.locks.indexOf(exports.LockOptions.RANDOM) !== -1) {
                        setTimeout(function () {
                            _this.locks[i] = true;
                        }, Math.random() * 1000 * 60);
                    }
                }
                this.lockColors[i] = this.showLockIcon(i);
                this.lockIcons[i] = this.showLockIcon(i);
                this._onChange();
            }
        };
        NgxBadPhoneSpinnerComponent.prototype.randomizeDigit = function (i, increment) {
            var _this = this;
            this.isRandomizing = true;
            this.digits[i] = Math.round(Math.random() * 9);
            if (typeof increment === 'undefined') {
                increment = this.options.shufflesPerClick;
                increment--;
            }
            if (increment !== 0) {
                increment--;
                setTimeout(function () {
                    _this.randomizeDigit(i, increment);
                }, this.options.shuffleMilliseconds);
            }
            else {
                this.isRandomizing = false;
                var shuffleLock = this.options.locks.indexOf(exports.LockOptions.SHUFFLE) !== -1;
                var shuffleUnlock = this.options.unlocks.indexOf(exports.LockOptions.SHUFFLE) !== -1;
                if (shuffleLock || shuffleUnlock) {
                    for (var i_1 = 0; i_1 < 10; i_1++) {
                        if (this.locks[i_1] === false && shuffleLock) {
                            this.locks[i_1] = true;
                        }
                        if (this.locks[i_1] === true && shuffleUnlock) {
                            this.locks[i_1] = false;
                        }
                        this.lockColors[i_1] = this.showLockIcon(i_1);
                        this.lockIcons[i_1] = this.showLockIcon(i_1);
                    }
                }
            }
        };
        NgxBadPhoneSpinnerComponent.prototype.clearLocks = function () {
            for (var i = 0; i < 10; i++) {
                this.locks[i] = false;
                this.lockColors[i] = this.showLockIcon(i);
                this.lockIcons[i] = this.showLockIcon(i);
            }
        };
        NgxBadPhoneSpinnerComponent.prototype.reset = function () {
            this.clearLocks();
            this.randomize();
        };
        NgxBadPhoneSpinnerComponent.prototype.randomize = function () {
            for (var i = 0; i < 10; i++) {
                if (this.locks[i] === false) {
                    this.randomizeDigit(i);
                }
            }
        };
        NgxBadPhoneSpinnerComponent.prototype.showLockIcon = function (i) {
            var showLockIcon = this.locks[i];
            if (this.locks[i] === true && this.options.locks.indexOf(exports.LockOptions.LIAR) !== -1) {
                if (Math.random() < 0.25) {
                    showLockIcon = !this.locks[i];
                }
            }
            else if (this.locks[i] === false && this.options.unlocks.indexOf(exports.LockOptions.LIAR) !== -1) {
                if (Math.random() < 0.25) {
                    showLockIcon = !this.locks[i];
                }
            }
            return showLockIcon;
        };
        NgxBadPhoneSpinnerComponent.prototype._onChange = function () {
            this.fullNumber = '';
            for (var i = 0; i < 10; i++) {
                this.fullNumber += '' + this.digits[i];
            }
            this._propagateChange(this.fullNumber);
            this.change.emit(this.fullNumber);
        };
        var NgxBadPhoneSpinnerComponent_1;
        __decorate([
            core.Input()
        ], NgxBadPhoneSpinnerComponent.prototype, "options", void 0);
        __decorate([
            core.Input()
        ], NgxBadPhoneSpinnerComponent.prototype, "number", null);
        __decorate([
            core.Output()
        ], NgxBadPhoneSpinnerComponent.prototype, "change", void 0);
        NgxBadPhoneSpinnerComponent = NgxBadPhoneSpinnerComponent_1 = __decorate([
            core.Component({
                selector: 'ngx-bad-phone-spinner',
                template: "<div class=\"ngx-bad-phone-spinner\">\n  <div class=\"inline-label\">(</div>\n\n  <div class=\"wrapper\"\n       *ngFor=\"let range of [] | range:3; let i = index\">\n    <div>{{ digits[i] }}</div>\n\n    <button type=\"button\"\n            class=\"ripple\"\n            (click)=\"toggleLock(i)\"\n            [disabled]=\"disabled || isRandomizing\"\n            [ngClass]=\"{ 'locked': locks[i], 'unlocked': locks[i] }\">\n      <span *ngIf=\"lockIcons[i] === true\">\n        <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"-150 -100 700 700\" [attr.fill]=\"lockColors[i] ? 'red' : 'green'\">\n          <path d=\"M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z\"/>\n        </svg>\n      </span>\n\n      <span *ngIf=\"lockIcons[i] === false\">\n        <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"-150 -100 700 700\" [attr.fill]=\"lockColors[i] ? 'red' : 'green'\">\n          <path d=\"M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z\"/>\n        </svg>\n      </span>\n    </button>\n  </div>\n\n  <div class=\"inline-label\">)</div>\n\n  <div class=\"wrapper\"\n       *ngFor=\"let range of [] | range:3; let i = index\">\n    <div>{{ digits[i + 3] }}</div>\n\n    <button type=\"button\"\n            class=\"ripple\"\n            (click)=\"toggleLock(i + 3)\"\n            [disabled]=\"disabled || isRandomizing\"\n            [ngClass]=\"{ 'locked': locks[i + 3], 'unlocked': locks[i + 3] }\">\n      <span *ngIf=\"lockIcons[i + 3] === true\">\n        <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"-150 -100 700 700\" [attr.fill]=\"lockColors[i + 3] ? 'red' : 'green'\">\n          <path d=\"M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z\"/>\n        </svg>\n      </span>\n\n      <span *ngIf=\"lockIcons[i + 3] === false\">\n        <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"-150 -100 700 700\" [attr.fill]=\"lockColors[i + 3] ? 'red' : 'green'\">\n          <path d=\"M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z\"/>\n        </svg>\n      </span>\n    </button>\n  </div>\n\n  <div class=\"inline-label\">-</div>\n\n  <div class=\"wrapper\"\n       *ngFor=\"let range of [] | range:4; let i = index\">\n    <div>{{ digits[i + 6] }}</div>\n\n    <button type=\"button\"\n            class=\"ripple\"\n            (click)=\"toggleLock(i + 6)\"\n            [disabled]=\"disabled || isRandomizing\"\n            [ngClass]=\"{ 'locked': locks[i + 6], 'unlocked': locks[i+ 6] }\">\n      <span *ngIf=\"lockIcons[i + 6] === true\">\n        <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"-150 -100 700 700\" [attr.fill]=\"lockColors[i + 6] ? 'red' : 'green'\">\n          <path d=\"M400 224h-24v-72C376 68.2 307.8 0 224 0S72 68.2 72 152v72H48c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V272c0-26.5-21.5-48-48-48zm-104 0H152v-72c0-39.7 32.3-72 72-72s72 32.3 72 72v72z\"/>\n        </svg>\n      </span>\n\n      <span *ngIf=\"lockIcons[i + 6] === false\">\n        <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"-150 -100 700 700\" [attr.fill]=\"lockColors[i + 6] ? 'red' : 'green'\">\n          <path d=\"M400 256H152V152.9c0-39.6 31.7-72.5 71.3-72.9 40-.4 72.7 32.1 72.7 72v16c0 13.3 10.7 24 24 24h32c13.3 0 24-10.7 24-24v-16C376 68 307.5-.3 223.5 0 139.5.3 72 69.5 72 153.5V256H48c-26.5 0-48 21.5-48 48v160c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V304c0-26.5-21.5-48-48-48z\"/>\n        </svg>\n      </span>\n    </button>\n  </div>\n\n  <button type=\"button\"\n          class=\"randomize-button ripple\"\n          (click)=\"randomize()\"\n          [disabled]=\"disabled || isRandomizing\">\n    <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 512 512\"><path d=\"M504.971 359.029c9.373 9.373 9.373 24.569 0 33.941l-80 79.984c-15.01 15.01-40.971 4.49-40.971-16.971V416h-58.785a12.004 12.004 0 0 1-8.773-3.812l-70.556-75.596 53.333-57.143L352 336h32v-39.981c0-21.438 25.943-31.998 40.971-16.971l80 79.981zM12 176h84l52.781 56.551 53.333-57.143-70.556-75.596A11.999 11.999 0 0 0 122.785 96H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12zm372 0v39.984c0 21.46 25.961 31.98 40.971 16.971l80-79.984c9.373-9.373 9.373-24.569 0-33.941l-80-79.981C409.943 24.021 384 34.582 384 56.019V96h-58.785a12.004 12.004 0 0 0-8.773 3.812L96 336H12c-6.627 0-12 5.373-12 12v56c0 6.627 5.373 12 12 12h110.785c3.326 0 6.503-1.381 8.773-3.812L352 176h32z\"></path></svg>\n  </button>\n\n  <button type=\"button\"\n          class=\"randomize-button ripple\"\n          (click)=\"clearLocks()\"\n          [disabled]=\"disabled || isRandomizing\">\n    <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 512 512\"><path d=\"M512 176.001C512 273.203 433.202 352 336 352c-11.22 0-22.19-1.062-32.827-3.069l-24.012 27.014A23.999 23.999 0 0 1 261.223 384H224v40c0 13.255-10.745 24-24 24h-40v40c0 13.255-10.745 24-24 24H24c-13.255 0-24-10.745-24-24v-78.059c0-6.365 2.529-12.47 7.029-16.971l161.802-161.802C163.108 213.814 160 195.271 160 176 160 78.798 238.797.001 335.999 0 433.488-.001 512 78.511 512 176.001zM336 128c0 26.51 21.49 48 48 48s48-21.49 48-48-21.49-48-48-48-48 21.49-48 48z\"/></svg>\n  </button>\n\n  <button type=\"button\"\n          class=\"randomize-button ripple\"\n          (click)=\"reset()\"\n          [disabled]=\"disabled || isRandomizing\">\n    <svg xmlns=\"http://www.w3.org/2000/svg\" viewBox=\"0 0 512 512\"><path d=\"M500.33 0h-47.41a12 12 0 0 0-12 12.57l4 82.76A247.42 247.42 0 0 0 256 8C119.34 8 7.9 119.53 8 256.19 8.1 393.07 119.1 504 256 504a247.1 247.1 0 0 0 166.18-63.91 12 12 0 0 0 .48-17.43l-34-34a12 12 0 0 0-16.38-.55A176 176 0 1 1 402.1 157.8l-101.53-4.87a12 12 0 0 0-12.57 12v47.41a12 12 0 0 0 12 12h200.33a12 12 0 0 0 12-12V12a12 12 0 0 0-12-12z\"/></svg>\n  </button>\n</div>\n",
                providers: [
                    {
                        provide: forms.NG_VALUE_ACCESSOR,
                        useExisting: core.forwardRef(function () { return NgxBadPhoneSpinnerComponent_1; }),
                        multi: true
                    }
                ],
                styles: ["input[type=number]::-webkit-inner-spin-button,input[type=number]::-webkit-outer-spin-button{-webkit-appearance:none;margin:0}.ngx-bad-phone-spinner{display:flex;width:100%}.ngx-bad-phone-spinner .wrapper{border-top:1px solid rgba(0,0,0,.12);float:left;max-width:50px;padding:1px}.ngx-bad-phone-spinner .wrapper button{padding:.5rem;overflow:hidden;border:0}.ngx-bad-phone-spinner .wrapper button:last-child{border:1px solid rgba(0,0,0,.12);border-radius:0 0 10px 10px;width:100%}.ngx-bad-phone-spinner .wrapper button:focus{outline:0}.ngx-bad-phone-spinner .wrapper button:disabled{cursor:not-allowed}.ngx-bad-phone-spinner .wrapper button:disabled svg{fill:rgba(0,0,0,.26)}.ngx-bad-phone-spinner .wrapper div{align-items:center;border-left:1px solid rgba(0,0,0,.12);border-right:1px solid rgba(0,0,0,.12);padding:.5rem;font-size:1rem;flex:1;min-width:20px;text-align:center;vertical-align:unset}.ngx-bad-phone-spinner .inline-label{float:left;padding-top:0;padding-left:5px;padding-right:5px;font-size:2em}.ngx-bad-phone-spinner .ripple{background-position:center;transition:background .3s}.ngx-bad-phone-spinner .ripple:hover{cursor:pointer;background:radial-gradient(circle,transparent 1%,rgba(0,0,0,.12) 1%) center/15000% rgba(0,0,0,.12)}.ngx-bad-phone-spinner .ripple:active{background-color:rgba(0,0,0,.12);background-size:100%;transition:background}.ngx-bad-phone-spinner .randomize-button{width:45px;border-radius:9px;margin-left:10px;height:37px}"]
            })
        ], NgxBadPhoneSpinnerComponent);
        return NgxBadPhoneSpinnerComponent;
    }());

    var RangePipe = /** @class */ (function () {
        function RangePipe() {
        }
        RangePipe.prototype.transform = function (items, quantity) {
            items.length = 0;
            for (var i = 0; i < quantity; i++) {
                items.push(i);
            }
            return items;
        };
        RangePipe = __decorate([
            core.Pipe({
                name: 'range',
                pure: false
            })
        ], RangePipe);
        return RangePipe;
    }());

    var NgxBadPhoneSpinnerModule = /** @class */ (function () {
        function NgxBadPhoneSpinnerModule() {
        }
        NgxBadPhoneSpinnerModule_1 = NgxBadPhoneSpinnerModule;
        NgxBadPhoneSpinnerModule.forRoot = function () {
            return {
                ngModule: NgxBadPhoneSpinnerModule_1,
                providers: []
            };
        };
        var NgxBadPhoneSpinnerModule_1;
        NgxBadPhoneSpinnerModule = NgxBadPhoneSpinnerModule_1 = __decorate([
            core.NgModule({
                declarations: [
                    NgxBadPhoneSpinnerComponent,
                    RangePipe
                ],
                exports: [
                    NgxBadPhoneSpinnerComponent,
                    RangePipe
                ],
                imports: [
                    common.CommonModule,
                    forms.FormsModule
                ],
                providers: []
            })
        ], NgxBadPhoneSpinnerModule);
        return NgxBadPhoneSpinnerModule;
    }());

    exports.NgxBadPhoneSpinnerComponent = NgxBadPhoneSpinnerComponent;
    exports.NgxBadPhoneSpinnerModule = NgxBadPhoneSpinnerModule;
    exports.NgxBadPhoneSpinnerOptions = NgxBadPhoneSpinnerOptions;
    exports.ɵa = RangePipe;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
