// ngx-bad-phone-spinner/esm5/lib/ngx-bad-phone-spinner.model.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngx-bad-phone-spinner.model.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var NgxBadPhoneSpinnerOptions = /** @class */ (function () {
    function NgxBadPhoneSpinnerOptions() {
        this.locks = [
            LockOptions.MANUAL
        ];
        this.unlocks = [
            LockOptions.MANUAL
        ];
        this.shuffleMilliseconds = 200;
        this.shufflesPerClick = 5;
    }
    return NgxBadPhoneSpinnerOptions;
}());
export { NgxBadPhoneSpinnerOptions };
if (false) {
    /** @type {?} */
    NgxBadPhoneSpinnerOptions.prototype.locks;
    /** @type {?} */
    NgxBadPhoneSpinnerOptions.prototype.unlocks;
    /** @type {?} */
    NgxBadPhoneSpinnerOptions.prototype.shuffleMilliseconds;
    /** @type {?} */
    NgxBadPhoneSpinnerOptions.prototype.shufflesPerClick;
}
/** @enum {string} */
var LockOptions = {
    IGNORE: "ignore",
    LIAR: "liar",
    MANUAL: "manual",
    ORDER: "order",
    RANDOM: "random",
    REVERSE: "reverse",
    SHUFFLE: "shuffle",
};
export { LockOptions };
