// ngx-bad-phone-spinner/esm5/lib/ngx-bad-phone-spinner.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ngx-bad-phone-spinner.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { CommonModule } from '@angular/common';
import { FormsModule } from '@angular/forms';
import { NgxBadPhoneSpinnerComponent } from './ngx-bad-phone-spinner/ngx-bad-phone-spinner.component';
import { RangeModule } from './range/range.module';
var NgxBadPhoneSpinnerModule = /** @class */ (function () {
    function NgxBadPhoneSpinnerModule() {
    }
    NgxBadPhoneSpinnerModule.decorators = [
        { type: NgModule, args: [{
                    declarations: [
                        NgxBadPhoneSpinnerComponent
                    ],
                    entryComponents: [
                        NgxBadPhoneSpinnerComponent
                    ],
                    exports: [
                        NgxBadPhoneSpinnerComponent
                    ],
                    imports: [
                        BrowserModule,
                        CommonModule,
                        FormsModule,
                        RangeModule
                    ],
                    providers: []
                },] }
    ];
    return NgxBadPhoneSpinnerModule;
}());
export { NgxBadPhoneSpinnerModule };
